backend/amd64/emit.mlp

```diff
@@ -1475,10 +1478,56 @@ let emit_probe_notes0 () =
       add_def_symbol label)
     !probe_semaphores
 
-let emit_probe_notes () =
-  match !probes with
-  | [] -> ()
-  | _ -> emit_probe_notes0 ()
+let emit_probe_notes probes =
+  match probes, String.Map.is_empty !probe_semaphores with
+  | [], true -> ()
+  | _ -> emit_probe_notes0 probes
+
+let emit_dummy_probe_sites () =
+  let semaphores_without_probes =
+    List.fold_left (fun acc probe ->
+      String.Map.remove probe.probe_name acc)
+      !probe_semaphores !probes in
+  if (String.Map.is_empty semaphores_without_probes) then []
+  else begin
+    let fun_name = Compilenv.make_symbol (Some "___dummy_probes") in
+
+    (* start the function (as in fundecl) *)
+    emit_named_text_section fun_name;
+    D.align 16;
+    add_def_symbol fun_name;
+    D.global (emit_symbol fun_name);
+    D.label (emit_symbol fun_name);
+    cfi_startproc ();
+
+    (* emit dummy probe sites: labeled NOP only for stapsdt probes to work,
+       but no calls to probe wrapper because there is no ocaml probe handler. *)
+    let dummy_probes =
+      String.Map.fold (fun probe_name _label acc ->
+        let probe_label = new_label () in
+        def_label probe_label;
+        I.nop (); (* for uprobes and usdt probes as well *)
+        let probe = { probe_name;
+                      probe_label;
+                      probe_arg = [||];
+                      probe_live = Reg.Set.empty;
+                      probe_handler_code_sym = None;
+                      (* the following fields are unused for dummy probes,
+                         which have no ocaml handler and do not call
+                         probe wrappers. *)
+                      probe_stack_offset = !stack_offset;
+                      probe_num_stack_slots = num_stack_slots;
```

Comment:
(If they are not used, maybe give them nonsensical values.
Another possibility would be to use different types, but that
seems overkill.)

Reply:
Yes, proper types seem to over-complicate the rest of the probes emitter code for a minor case.  I can't think of a "safe" invalid value for stack_offset, so I'll the code as is for now.

Reply:
(My point was precisely an unsafe/notable value, so that
any mistake/misuse is obvious; a `deadbeef`-like value
if you will.)